cdap-docs/admin-manual/source/operations/logging.rst

```diff
@@ -10,35 +10,106 @@ Logging and Monitoring
+application may similarly be distributed.
 
+To address these issues, the CDAP log framework was designed:
 
-Log Location
-============
-The location of logs depends on the mode of CDAP and the Hadoop distribution.
+- to centralize the location of logs, so that the individual logs of an application's programs can be
+  merged into one;
 
-- With **Standalone CDAP** (the CDAP SDK), logs are located in ``<CDAP-SDK-HOME>/logs``.
+- to make logs persistent and available for later use and analysis; and
 
-- For **Distributed CDAP**, with the exception of Cloudera Manager, logs are located in
-  ``/var/log/cdap``. With Cloudera Manager installations, log files are located in
-  directories under ``/var/run/cloudera-scm-agent/process``.
+- to use a pluggable system that can be adapted for individual requirements at both the
+  application and cluster levels.
 
 
-Logging Messages
-================
+Logging Framework
+=================
+
+.. highlight:: console
+
+This diagram shows the steps CDAP follows with its logging framework:
+
+  1. Logs are collected from individual programs running in YARN containers.
+
+  #. YARN containers have their own local log files.
+
+  #. CDAP logs are published to Kafka topics.
+
+  #. The Kafka topics are consumed by the Log Saver Service. Each instance of the service
+     has one or more logging pipelines. Each pipeline buffers log messages in-memory and
+     sorts them based on their timestamp. Custom pipelines can be configured by adding an
+     XML file in a prescribed location.
+   
+  #. Metrics are created based on the results of the pipelines, and passed to the Metrics Service.
+   
+  #. Logs are written out to persistent storage in HDFS by the logging pipelines.
+
+.. figure:: /_images/logging-framework.png
+    :figwidth: 100%
+    :width: 800px
+    :align: center
+
+    **CDAP Logging Framework:** From YARN containers, through Kafka and the Log Saver Service, to HDFS
+    
+The actual logging uses standard `SLF4J (Simple Logging Facade for Java)
+<http://www.slf4j.org/manual.html>`__ APIs and `Logback
+<https://logback.qos.ch/manual>`__, and consists of *logging pipelines* with *appenders*:
+
+  - A **logging pipeline** is a process that consumes log events from Kafka, invokes
+    an *appender* defined in its configuration, buffering, sorting, and then persisting the log messages.
+
+  - An **appender** is a Java class, responsible for writing the log events to persistent storage and
+    maintaining metadata about that storage.
+
+The *Logging Framework* is configured using Logback's ``logback.xml`` files:
+
+- The ``logback-container.xml`` controls the program containers, both locally and to Kafka.
+- The ``cdap-site.xml`` controls the CDAP Log Pipeline for CDAP System Services, and the
+  reading, buffering, and memory allocated.
+- A ``custom-pipeline.xml``, if present, triggers the creation of a custom pipeline, with
```

Comment:
there can be any number of `xml` files with any name, this description sound like we look for this name.